.github/type_check.sh

```diff
@@ -0,0 +1,5 @@
+if [ "$(basename "$(pwd)")" = ".github" ]; then
+    cd ..
+fi
+
+pyright -p .github/pyright-config.json
```

Comment:
Do you feel confident in porting this to .bat for our windows users? While awful from a maintenance standpoint, I know of at least one regular contributor that does not have any bash installed. You can test the .bat without running Windows by switching `runs-on` to `windows-latest` in the gh action.

Reply:
How about this change.
Now it uses python instead of bash or bat.
I think that will be better if we add mypy anyway.